cpp/CMakeLists.txt

```diff
@@ -13,14 +13,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #=============================================================================
-cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
```

Comment:
If we're bumping the minimum to 3.18 we should probably set policy CMP0104 to `OLD` until we use the proper `CMAKE_CUDA_ARCHITECTURES` / `CUDA_ARCHITECTURES` options.

https://cmake.org/cmake/help/v3.18/policy/CMP0104.html

Reply:
I'm not sure what this means

Reply:
`CMP0104` is a CMake policy that was added in CMake 3.18 that automatically sets the `CUDA_ARCHITECTURES` property of all targets to `CMAKE_CUDA_ARCHITECTURES` which if not set by the user, defaults to the oldest supported architecture of nvcc (Maxwell, 52 for CUDA 11.0).

By setting the policy to `OLD` it reverts to the pre CMake 3.18 behavior where `CUDA_ARCHITECTURES` is not implicitly set, so we can continue to use the nvcc compiler flags as we're currently doing.

Reply:
I'm not familiar with cmake policies. How do I set it to OLD?

Reply:
`cmake_policy(CMP0104 OLD)`